gallery_dl/extractor/instagram.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 Leonardo Taccari
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extract images from Instagram"""
+
+import hashlib
+import json
+from .common import Extractor, Message
+from .. import text
+
+
+class InstagramExtractor(Extractor):
+    """Base class for instagram extractors"""
+    category = "instagram"
+    directory_fmt = ["{category}", "{username}"]
+    filename_fmt = "{media_id}.{extension}"
+    archive_fmt = "{media_id}"
+    root = "https://www.instagram.com"
+
+    def __init__(self, match):
+        Extractor.__init__(self)
```

Comment:
This appears rather pointless. Don't use an `__init__` function if there is nothing to do.

Reply:
I have removed it and adjusted the other `__init__()` to call
`InstagramExtractor.__init__(self)` instead of
`InstagramExtractor.__init__(self, match)`.